File: app/src/main/java/com/MohamedTaha/Imagine/New/dagger2/module/ActivityModule.java
package com.MohamedTaha.Imagine.New.dagger2.module;

import android.app.Activity;

import com.MohamedTaha.Imagine.New.scope.ScopeFragment;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
  //  @ScopeFragment
    @Provides
    public Activity getActivity() {
        return new Activity();
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/hilt/module/ActivityModule.kt
package com.mohamedtaha.imagine.hilt.module

import android.os.Bundle
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent
import java.util.*

@Module
@InstallIn(ActivityComponent::class)
class ActivityModule {
    @Provides
    fun timerProvide():Timer = Timer()

    @Provides
    fun bundleProvide():Bundle = Bundle()

}